### PR TITLE
chore(rate-limiting): clarify rate-limiting accuracy

### PR DESCRIPTION
**Pull request by nowNick**
### Description

Both rate-limiting and rate-limiting-advanced plugins documentation specify that various policies are possible and states that `cluster` and `redis` are "Accurate". However it's possible to configure a `sync_rate` option for these plugins and set it to a value that does not offer synchronous behaviour allowing for some slippage in favor of performance.

This commit clarifies that `sync_rate` affects mentioned accuracy.

KAG-2896

https://github.com/Kong/kong/issues/11846

### Testing instructions

Preview link:
- [rate-limiting](https://deploy-preview-7607--kongdocs.netlify.app/hub/kong-inc/rate-limiting/#strategies)
- [rate-limiting-advanced](https://deploy-preview-7607--kongdocs.netlify.app/hub/kong-inc/rate-limiting-advanced/#strategies)

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md [hunk 1]
```diff
@@ -155,8 +155,10 @@ The plugin supports three strategies.
 | Strategy    | Pros | Cons   |
 | --------- | ---- | ------ |
 | `local`   | Minimal performance impact. | Less accurate. Unless there's a consistent-hashing load balancer in front of Kong, it diverges when scaling the number of nodes.
-| `cluster` | Accurate, no extra components to support. | Each request forces a read and a write on the data store. Therefore, relatively, the biggest performance impact. |
-| `redis`   | Accurate, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
+| `cluster` | Accurate*, no extra components to support. | Each request forces a read and a write on the data store. Therefore, relatively, the biggest performance impact. |
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md:158]:
```suggestion
| `cluster` | Accurate<sup>1</sup>, no extra components to support. | Each request forces a read and a write on the data store. Therefore, relatively, the biggest performance impact. |
```

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md [hunk 2]
```diff
@@ -155,8 +155,10 @@ The plugin supports three strategies.
...
+| `redis`   | Accurate*, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md:159]:
```suggestion
| `redis`   | Accurate<sup>1</sup>, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
```

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md [hunk 3]
```diff
@@ -155,8 +155,10 @@ The plugin supports three strategies.
...
+
+**only when `sync_rate` option is set to `0` (synchronous behaviour). Please read the config reference for more details.*
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md:161]:
```suggestion
{:.note .no-icon}
> **\[1\]**: Only when `sync_rate` option is set to `0` (synchronous behavior). See the [configuration reference](/hub/kong-inc/rate-limiting-advanced/configuration/#config-sync_rate) for more details.
```

File: app/_hub/kong-inc/rate-limiting/overview/_index.md [hunk 4]
```diff
@@ -68,7 +68,9 @@ The plugin supports three strategies.
 | --------- | ---- | ------ |
 | `local`   | Minimal performance impact. | Less accurate. Unless there's a consistent-hashing load balancer in front of Kong, it diverges when scaling the number of nodes.
 | `cluster` | Accurate, no extra components to support. | Each request forces a read and a write on the data store. Therefore, relatively, the biggest performance impact. |
-| `redis`   | Accurate, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
+| `redis`   | Accurate*, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
+
+**only when `sync_rate` option is set to `-1` (synchronous behaviour). When using Redis it's possible to set it to a positive value which might lead to some inaccuracies. Please read the config reference for more details.*
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting/overview/_index.md:73]:
```suggestion
{:.note .no-icon}
> **\[1\]**: Only when `sync_rate` option is set to `-1` (synchronous behavior). When using Redis, it's possible to set the sync rate to a positive value, which can lead to some inaccuracies. See the [configuration reference](/hub/kong-inc/rate-limiting/configuration/#config-sync_rate) for more details.
```

File: app/_hub/kong-inc/rate-limiting/overview/_index.md [hunk 5]
```diff
@@ -68,7 +68,9 @@ The plugin supports three strategies.
 | --------- | ---- | ------ |
 | `local`   | Minimal performance impact. | Less accurate. Unless there's a consistent-hashing load balancer in front of Kong, it diverges when scaling the number of nodes.
 | `cluster` | Accurate, no extra components to support. | Each request forces a read and a write on the data store. Therefore, relatively, the biggest performance impact. |
-| `redis`   | Accurate, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
+| `redis`   | Accurate*, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
```
Comment (lena-larionova) [inline app/_hub/kong-inc/rate-limiting/overview/_index.md:71]:
```suggestion
| `redis`   | Accurate<sup>1</sup>, less performance impact than a `cluster` policy. | Needs a Redis installation. Bigger performance impact than a `local` policy. |
```

**Review by samugi — APPROVED** [review]

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md (hunk 1 above)
Comment (nowNick) [reply app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md]: applied suggestion ✅

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md (hunk 2 above)
Comment (nowNick) [reply app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md]: applied suggestion ✅

File: app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md (hunk 3 above)
Comment (nowNick) [reply app/_hub/kong-inc/rate-limiting-advanced/overview/_index.md]: applied suggestion ✅

File: app/_hub/kong-inc/rate-limiting/overview/_index.md (hunk 4 above)
Comment (nowNick) [reply app/_hub/kong-inc/rate-limiting/overview/_index.md]: applied suggestion ✅

File: app/_hub/kong-inc/rate-limiting/overview/_index.md (hunk 5 above)
Comment (nowNick) [reply app/_hub/kong-inc/rate-limiting/overview/_index.md]: applied suggestion ✅

**Comment by nowNick** [comment]
Thank you @lena-larionova for your suggestions! I've applied all of them. Please take a look :)

**Review by lena-larionova — APPROVED** [review]
LGTM! Thanks for the clarification.

**Pull request merged**